src/mac/command.rs

```diff
@@ -0,0 +1,817 @@
+    DisassociationNotification => 3,
+    /// Data request
+    DataRequest => 4,
+    /// PAN identifier conflict notification, sent from coordinator to offending device
+    PanIdConflictNotification => 5,
+    /// Orphan notification, 
+    OrphanNotification => 6,
+    /// Beacon request, sent from a device which want to join a PAN
+    BeaconRequest => 7,
+    /// Coordinator re-alignment, the coordinator will change network parameters
+    CoordinatorRealignment => 8,
+    /// Guaranteed time slot request, request a guaranteed time slot
+    GuaranteedTimeSlotRequest => 9,
+);
+
+const CAP_FFD: u8 = 0x02;
+const CAP_MAINS_POWER: u8 = 0x04;
+const CAP_IDLE_RECEIVE: u8 = 0x08;
+const CAP_FRAME_PROTECTION: u8 = 0x40;
+const CAP_ALLOCATE_ADDRESS: u8 = 0x80;
+
+/// Association request capability information
+///
+/// Sent with association request to report the capabilities of the device.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct CapabilityInformation {
+    /// Full-function device (FFD) or a reduced-function device (RFD)
+    /// RFD and FFD have different function sets.
+    pub full_function_device: bool,
+    /// Device is connected to a mains power source or not
+    pub mains_power: bool,
+    /// Receive is enabled while idle
+    pub idle_receive: bool,
+    /// Frames are cryptographically protected
+    pub frame_protection: bool,
+    /// Device wish to have an short address allocated by the coordinator
+    pub allocate_address: bool,
+}
+
+impl From<u8> for CapabilityInformation {
+    fn from(byte: u8) -> Self {
+        let full_function_device = byte & CAP_FFD == CAP_FFD;
+        let mains_power = byte & CAP_MAINS_POWER == CAP_MAINS_POWER;
+        let idle_receive = byte & CAP_IDLE_RECEIVE == CAP_IDLE_RECEIVE;
+        let frame_protection = byte & CAP_FRAME_PROTECTION == CAP_FRAME_PROTECTION;
+        let allocate_address = byte & CAP_ALLOCATE_ADDRESS == CAP_ALLOCATE_ADDRESS;
+        Self {
+            full_function_device,
+            mains_power,
+            idle_receive,
+            frame_protection,
+            allocate_address,
+        }
+    }
+}
+
+impl From<CapabilityInformation> for u8 {
+    fn from(ar: CapabilityInformation) -> Self {
+        let mut byte = 0u8;
+        if ar.full_function_device {
+            byte = byte | CAP_FFD;
+        }
+        if ar.mains_power {
+            byte = byte | CAP_MAINS_POWER;
+        }
+        if ar.idle_receive {
+            byte = byte | CAP_IDLE_RECEIVE;
+        }
+        if ar.frame_protection {
+            byte = byte | CAP_FRAME_PROTECTION;
+        }
+        if ar.allocate_address {
+            byte = byte | CAP_ALLOCATE_ADDRESS;
+        }
+        byte
+    }
+}
+
+extended_enum!(
+    /// Association Status
+    AssociationStatus, u8,
+    /// Successful
+    Successful => 0x00,
+    /// Network (PAN) at capacity
+    NetworkAtCapacity => 0x01,
+    /// Access to PAN denied
+    AccessDenied => 0x02,
+    /// ???
```

Comment:
This comment is not very informative :-)

Reply:
No, I will come up with something better.